### PR TITLE
DDCE-5271 - Add new API 393 and 394 for getting benefits and updating…

### PR DESCRIPTION
**Pull request by frikit**
… benefits

### CODE REVIEW

File: app/controllers/actions/MinimalAuthAction.scala [hunk 1]
```diff
@@ -20,30 +20,41 @@ import com.google.inject.ImplementedBy
...
+        case Some(value) =>
+          val userPid: String = value.providerId
+          Future.successful(Right(AuthenticatedRequest(request, userPid)))
+
+        case None =>
+          throw new RuntimeException("No user PID found")
+      }
+      .recover { case t: Throwable =>
+        logger.debug("Debug info - " + t.getMessage)
+        Left(Unauthorized)
```
Comment (lmugnai) [inline app/controllers/actions/MinimalAuthAction.scala:52]: coverage report is showing missing tests for this. Could you take a look?

File: app/controllers/utils/ControllerUtils.scala [hunk 2]
```diff
@@ -104,26 +144,26 @@ class ControllerUtils @Inject() (configuration: Configuration)(implicit val exec
...
 
-  def getNPSMutatorSessionHeader(implicit request: Request[AnyContent]): Map[String, String] =
+  def getNPSMutatorSessionHeader(implicit request: Request[_]): Map[String, String] =
     if (request.headers.get(HeaderTags.ETAG).isDefined) {
-      Map(
-        (HeaderTags.ETAG, request.headers.get(HeaderTags.ETAG).getOrElse("0")),
-        (HeaderTags.X_TXID, request.headers.get(HeaderTags.X_TXID).getOrElse("1"))
-      )
+      val etag = request.headers.get(HeaderTags.ETAG).getOrElse(HeaderTags.ETAG_DEFAULT_VALUE)
+      val txid = request.headers.get(HeaderTags.X_TXID).getOrElse(HeaderTags.X_TXID_DEFAULT_VALUE)
```
Comment (lmugnai) [inline app/controllers/utils/ControllerUtils.scala:161]: coverage report is showing the getOrElse is not covered by tests. COuld you take a look?

File: app/models/v1/PbikAction.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+package models.v1
+
+import models.v1
+import play.api.libs.json.{Format, Json}
+
+/*
+string (EnumPbikactv)
+Enum: "No Action" "Reinstate Payrolled Benefit In Kind (PBIK)" "Remove Payrolled Benefit In Kind (PBIK)"
+ */
```
Comment (lmugnai) [inline app/models/v1/PbikAction.scala:25]: can we remove this?

File: app/controllers/utils/ControllerUtils.scala (hunk 2 above)
Comment (frikit) [reply app/controllers/utils/ControllerUtils.scala]: changed logic more scala way now is covered

**Pull request merged**